pyartcd/pyartcd/pipelines/gen_assembly.py

```diff
@@ -0,0 +1,99 @@
+import sys
+
+import click
+from doozerlib import exectools
+from pyartcd import exceptions, github
+from pyartcd.runtime import Runtime
+from pyartcd.cli import cli, click_coroutine, pass_runtime
+
+
+class GenPipeline:
+    def __init__(self, runtime: Runtime, nightlies: str, custom: str, in_flight_prev: str, previous: str,
+                 build_version: str, assembly_name: str, build_number: str):
+        self.runtime = runtime
+        self.nightlies = nightlies
+        self.custom = custom
+        self.in_flight_prev = in_flight_prev
+        self.previous = previous
+        self.build_version = build_version
+        self.assembly_name = assembly_name
+        self.build_number = build_number
+
+    async def run(self):
+        # Create doozer cmd arguments
+        nightly_args = ""
+        for nightly in self.nightlies.split(","):
+            nightly_args += f" --nightly {nightly.strip()}"
+
+        cmd = f"--group openshift-{self.build_version} release:gen-assembly --name {self.assembly_name} from-releases {nightly_args}"
+        if self.custom:
+            cmd += ' --custom'
+            if (self.in_flight_prev and self.in_flight_prev != 'none') or self.previous:
+                print("ERROR: Specifying IN_FLIGHT_PREV or PREVIOUS for a custom release is not allowed.")
+                sys.exit(1)
+
+        else:
+            if self.in_flight_prev and self.in_flight_prev.lower() != 'none':
+                cmd += f" --in-flight {self.in_flight_prev}"
+
+            if self.previous:
+                for previous_data in self.previous.split(','):
+                    cmd += f" --previous {previous_data.strip()}"
+            else:
+                cmd += ' --auto-previous'
+
+        cmd = cmd.replace(" *\\\n *", " ").replace(" *\n *", " ").strip()  # Cleaning whitespace
+
+        # Run doozer command
+        output_all = exectools.cmd_gather(f"doozer --assembly=stream {cmd}")
```

Comment:
Use `pyartcd.exectools.cmd_assert_async`

Reply:
Isn't it `cmd_gather_async` since I need the result of the command? Since `cmd_assert_async` returns only `proc.returncode`

Reply:
Done!